pkg/dataprotection/utils/envvar.go

```diff
@@ -29,10 +29,15 @@ import (
 	dptypes "github.com/apecloud/kubeblocks/pkg/dataprotection/types"
 )
 
-func BuildEnvByCredential(pod *corev1.Pod, credential *dpv1alpha1.ConnectionCredential) []corev1.EnvVar {
+func BuildEnvByTarget(pod *corev1.Pod, credential *dpv1alpha1.ConnectionCredential, containerPort *dpv1alpha1.ContainerPort) []corev1.EnvVar {
 	var envVars []corev1.EnvVar
+	envVars = append(envVars, corev1.EnvVar{Name: dptypes.DPDBHost, Value: intctrlutil.BuildPodHostDNS(pod)})
+	if containerPort != nil {
+		envVars = append(envVars, corev1.EnvVar{Name: dptypes.DPDBPort, Value: strconv.Itoa(int(GetContainerPortByName(pod, containerPort.ContainerName, containerPort.PortName)))})
+	} else {
+		envVars = append(envVars, corev1.EnvVar{Name: dptypes.DPDBPort, Value: strconv.Itoa(int(GetPodFirstContainerPort(pod)))})
+	}
```

Comment:
This code is not equivalent to the previous one.

e.g. if if credential != nil, there'll be two dptypes.DPDBPort envVars  appended.